lib/internal/Magento/Framework/DB/Logger/LoggerProxy.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\DB\Logger;
+
+class LoggerProxy extends LoggerAbstract
+{
+    /**
+     * Logger alias param name
+     */
+    const PARAM_ALIAS = 'db_logger_mode';
+
+    /**
+     * Logger log all param name
+     */
+    const PARAM_LOG_ALL = 'db_logger_all';
+
+    /**
+     * Logger query time param name
+     */
+    const PARAM_QUERY_TIME = 'db_logger_query_time';
+
+    /**
+     * Logger call stack param name
+     */
+    const PARAM_CALL_STACK = 'db_logger_stack';
+
+    /**
+     * File logger alias
+     */
+    const LOGGER_ALIAS_FILE = 'file';
+
+    /**
+     * Quiet logger alias
+     */
+    const LOGGER_ALIAS_DISABLED = 'disabled';
+
+    /**
+     * @var LoggerAbstract
+     */
+    private $logger;
+
+    /**
+     * LoggerProxy constructor.
+     * @param File $file
+     * @param Quiet $quiet
+     * @param bool $loggerAlias
+     * @param bool $logAllQueries
+     * @param float $logQueryTime
+     * @param bool $logCallStack
+     */
+    public function __construct(
+        File $file,
```

Comment:
please avoid passing dependency which will never be used. It is better to pass Factory instead, which will create the right logger based on alias the first time logger is requested.